classes/Guest.php

```diff
@@ -157,6 +157,7 @@ protected function getBrowser($userAgent)
     protected function getOs($userAgent)
     {
         $osArray = [
+            'Windows 11' => 'Windows NT 11',
```

Comment:
One day we will have to try: 
https://docs.microsoft.com/en-us/microsoft-edge/web-platform/how-to-detect-win11

Reply:
I agree that looking for a string inside a string is kind of a simplistic solution.

Using User-Agent Hints as suggested on the link, instead of User-Agent strings (legacy), requires JavaScript. So it may be tricky to implement, I would think of injecting JS code from PHP somewhere in the HTML and then retrieve the data using `XMLHttpRequest()`.

However, you've got me doing some research and there is a PHP function called `get_browser()` which returns an array of parsed information. (It requires browcap PHP extension https://www.php.net/manual/en/function.get-browser.php and there were complains of it being too slow before PHP v7.1.1 according to https://www.php.net/ChangeLog-7.php)

If it works, it may be a cleaner solution. But I need some time to play around with this.

Reply:
We may also take a look at http://thadafinser.github.io/UserAgentParserComparison/v5/index.html

BrowCap-php package could be included
https://packagist.org/packages/browscap/browscap-php

Apparently it detects more Browsers, Operating systems and Device types within 0.03008 ms.

or PiwikDeviceDetector (https://packagist.org/packages/piwik/device-detector)

Which apparently is even faster (parses in 0.00434 ms)

Reply:
Yes this is something we can investigate later that's a good idea to use a dependency than redo the wheel

Reply:
@PierreRambaud Why not use the Matomo library : https://github.com/matomo-org/device-detector ?